File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] t = new int[3];

        while (true) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            t[0] = Integer.parseInt(st.nextToken());
            t[1]= Integer.parseInt(st.nextToken());
            t[2] = Integer.parseInt(st.nextToken());
            Arrays.sort(t);
            if (t[0] == 0 && t[1] == 0 && t[2] == 0)
                break;

            if (t[2] >= t[0] + t[1])
                System.out.println("Invalid");
            else if (t[0] == t[1] && t[1] == t[2] && t[2] == t[0])
                System.out.println("Equilateral");
            else if ((t[0] == t[1] && t[1] != t[2])
                    || (t[0] != t[1] && t[1] == t[2])
                    || (t[0] != t[1] && t[2] == t[0]))
                System.out.println("Isosceles");
            else if (t[0] != t[1] && t[1] != t[2] && t[2] != t[0])
                System.out.println("Scalene");
            else
                System.out.println("Invalid");
        }
    }
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.kt
fun main() {
    while (true) {
        val triangle = readln().split(" ").map { it.toInt() }
        if (triangle[0] == 0 && triangle[1] == 0 && triangle[2] == 0) return

        if (triangle.max() >= triangle.sum() - triangle.max()) {
            println("Invalid")
        } else if (triangle[0] == triangle[1] && triangle[1] == triangle[2]) {
            println("Equilateral")
        } else if (triangle[0] != triangle[1] && triangle[1] != triangle[2] && triangle[2] != triangle[0]) {
            println("Scalene")
        } else {
            println("Isosceles")
        }
    }
}